framework/di/Container.php

```diff
@@ -456,48 +457,68 @@ protected function resolveDependencies($dependencies, $reflection = null)
     }
 
     /**
-     * Invoke callback with resolved dependecies parameters.
+     * Resolve dependencies for a function, can use either named or numbered $params.
      * @param callable $callback callable to be invoked.
      * @param array $params callback paramater.
```

Comment:
Could be either regular array or array of parameter name => parameter value. In case of indexed array parameters are passed based on their position in the function signaure.